src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -150,4 +155,73 @@ class TimeseriesModelNew {
   }
 }
 
-export default TimeseriesModelNew
\ No newline at end of file
+const timeseriesCalculationPool = createCalculationPool({maxSimultaneous: 4, method: 'stack'})
+
+export interface TimeseriesData {
+  getChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => number[]
+  requestChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => void
+  numChannels: () => number
+  numTimepoints: () => number
+  getSampleRate: () => number
+  onDataSegmentSet: (handler: (ds_factor: number, t1: number, t2: number) => void) => void
+}
+
+const useTimeseriesData = (recordingObject: any, recordingInfo: RecordingInfo, hither: HitherContext): TimeseriesData | null => {
```

Comment:
So as I understand it, this is a custom hook that provides an interface to the time series data in a recording object. When called, the hook checks if a model exists and if the recording object & metadata (`recordingInfo`) are the same as the last call. If the model is defined and the ultimate data source is unchanged, just return the model persisted by the framework.

If the model does not exist, or if there's a new data source, then we define a new model of type `TimeseriesModelNew` which provides some data and some functionality. The model contains the sample rate, the channel count, a total frame count, and a segment size (presumably this is some sort of page size related to how much data can be downloaded from source at once, since it scales based on number of channels). The model also provides a function `onRequestDataSegment` which takes a downscaling factor and a segment number; when called, this function uses hither to asynchronously fetch the corresponding segment of the time series.

Did I miss anything?

Reply:
Sounds about right.